numpy/_core/src/multiarray/dragon4.c

```diff
@@ -1823,14 +1837,17 @@ FormatPositional(char *buffer, npy_uint32 bufferSize, BigInt *mantissa,
     if (digits_left > numWholeDigits + has_sign) {
         npy_int32 shift = digits_left - (numWholeDigits + has_sign);
         npy_int32 count = pos;
-
+        
+        /* provided digits_left is too large, can't create the string required*/
         if (count + shift > maxPrintLen) {
```

Comment:
This is similar to above `shift` is unbounded in size, so `count + shift` might overflow.  So rephrase as `count > maPrintLen - shift`.